runtime-modules/storage/src/data_directory.rs

```diff
@@ -342,7 +427,28 @@ decl_module! {
 }
 
 impl<T: Trait> Module<T> {
+    // Ensures quota limit constraint satisfied, returns delta.
+    fn ensure_quota_limit_constraint_satisfied(
```

Comment:
Very elegant :)

Reply:
Thank you, it was hard to come up with :) https://github.com/Joystream/joystream/pull/2122/commits/ff989c45bcd0b849082ec16ae25b9adbab5cd13a#diff-5f5bdbc581be08dfecff6870d8fcf9b767904a351d254a70cf35267e5465589bR526